test/unit/test_il_definitions.c

```diff
@@ -0,0 +1,382 @@
+// SPDX-FileCopyrightText: 2021 heersin <teablearcher@gmail.com>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_il.h>
+#include <rz_util.h>
+#include "minunit.h"
+
+static bool is_euqal_bv(RzILBitVector x, RzILBitVector y) {
+	bool equal = (rz_il_bv_cmp(x, y) == 0 ? true : false);
+	return equal;
+}
+
+static bool is_equal_bool(RzILBool x, RzILBool y) {
+	return x->b == y->b;
+}
+
+bool test_rzil_bv_init(void) {
+	// create a zero vector
+	RzILBitVector bits_42 = rz_il_bv_new0(42);
+	mu_assert("init 42-bits vector", bits_42 && (bits_42->len == 42));
+
+	// create by given ut32
+	RzILBitVector bits_32 = rz_il_bv_new_from_ut32(32, 100);
+	RzILBitVector bits_cmp = rz_il_bv_new0(32);
+	rz_il_bv_set(bits_cmp, 25, true);
+	rz_il_bv_set(bits_cmp, 26, true);
+	rz_il_bv_set(bits_cmp, 29, true);
```

Comment:
what's the order of the bits?
I thought that bv[0] == true meant bv.val = 1, bv[1] == true meant bv.val = 2, bv[2] == true meant bv.val = 4, etc.

This seems to be a different ordering, am I wrong?

Reply:
Judging from the other tests, it seems i'm right. However, i think this is counter intuitive. cc @thestr4ng3r @XVilka

Reply:
Yeah, it's a different ordering.bv[0] is the first bit when read from left to right.